.github/workflows/test_notebooks.yml

```diff
@@ -0,0 +1,47 @@
+name: Test notebooks
+
+on:
+    schedule:
+      - cron: '0 0 * * 6'  # Run at midnight on Saturdays
+    push:
+      branches:
+        - master
+    pull_request:
+      branches:
+        - master
+    workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+
+    steps:
+        - uses: actions/checkout@v2
+
+        - name: pip cache
+          uses: actions/cache@v2
+          with:
+            path: ~/.cache/pip
+            key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+
+
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@v1
+          with:
+            python-version: ${{ matrix.python-version }}
+
+        - name: Setup Graphviz
+          uses: ts-graphviz/setup-graphviz@v1
+
+        - name: Install Python packages from requirements.txt
+          run: |
+            pip install --upgrade pip
+            pip install -r docs/source/notebook_requirements.txt 
```

Comment:
If it wasn't clear, I'd also recommend removing the newly created `notebook_requirements.txt` file.

Reply:
That's what I started with. This strategy fails. Just did this again to humour you. It seems that build reqs, requirements needed for sphinx to produce html files and requirements needed for the notebooks to run are three different things. All that is listed in that .txt file landed there because the runner explicitly threw a name error.

Reply:
Yes, `nbval` was not included in the `test` dependencies in `setup.py`. See this PR I made to demonstrate. https://github.com/BasisResearch/chirho/pull/318